backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java

```diff
@@ -3,30 +3,34 @@
 import gov.cdc.usds.simplereport.api.InternalIdResolver;
 import gov.cdc.usds.simplereport.api.PersonNameResolver;
 import gov.cdc.usds.simplereport.api.model.ApiFacility;
+import gov.cdc.usds.simplereport.api.model.errors.NoDataLoaderFoundException;
 import gov.cdc.usds.simplereport.db.model.Facility;
 import gov.cdc.usds.simplereport.db.model.PatientPreferences;
 import gov.cdc.usds.simplereport.db.model.Person;
 import gov.cdc.usds.simplereport.db.model.TestEvent;
 import gov.cdc.usds.simplereport.db.model.auxiliary.TestResultDeliveryPreference;
-import gov.cdc.usds.simplereport.service.TestEventService;
 import graphql.kickstart.execution.context.GraphQLContext;
 import graphql.kickstart.tools.GraphQLResolver;
 import graphql.schema.DataFetchingEnvironment;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import org.dataloader.DataLoader;
 import org.dataloader.DataLoaderRegistry;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class PatientDataResolver
     implements GraphQLResolver<Person>, PersonNameResolver<Person>, InternalIdResolver<Person> {
 
-  @Autowired private TestEventService _testEventService;
+  public static final String LAST_TEST_DATA_LOADER = "patientLastTestLoader";
 
-  public TestEvent getLastTest(Person p) {
-    return _testEventService.getLastTestResultsForPatientPermRestricted(p);
+  public CompletableFuture<TestEvent> getLastTest(Person person, DataFetchingEnvironment dfe) {
+    DataLoaderRegistry registry = ((GraphQLContext) dfe.getContext()).getDataLoaderRegistry();
+    DataLoader<UUID, TestEvent> lastTestLoader = registry.getDataLoader(LAST_TEST_DATA_LOADER);
+    if (lastTestLoader != null) {
+      return lastTestLoader.load(person.getInternalId());
+    }
+    throw new NoDataLoaderFoundException(LAST_TEST_DATA_LOADER);
```

Comment:
Further opportunity for fun higher-order method extraction here. Not sure it's worthwhile...

Reply:
I considered this, briefly, but I think the method that would get factored out would require an unchecked cast suppression (like DataLoaderRegistry.load itself has) so I decided against it in favor of the few extra lines